File: Dante1/Module1.05/Zadanie5.11/main.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main()
{
    char input_text[1000];

    printf("Enter: ");
    scanf("%1000[^\n]s", input_text);

    for(size_t i = 0; i < strlen(input_text); i++)
    {
        if(!isdigit(input_text[i]) && input_text[i] != '|' && input_text[i] != '-')
        {
            printf("Incorrect input");
            return 1;
        }
    }

    long result[100];
    int result_index = 0;
    long sum = 0;

    int inside_segment = 0;
    long current_number = 0;
    int number_exists = 0;

    for(size_t i = 0; i <= strlen(input_text); i++)
    {
        char current = input_text[i];

        if (current == '|' || current == '\0')
        {
            if (inside_segment)
            {
                sum += current_number;
                result[result_index++] = sum;
                sum = 0;
                current_number = 0;
            }
            else
            {
                result[result_index++] = 0;
            }
            inside_segment = 1;
            number_exists = 0;
        }
        else if (isdigit(current))
        {
            current_number = current_number * 10 + (current - '0');
            number_exists = 1;
        }
        else if (current == '-')
        {
            if (number_exists)
            {
                sum = sum + current_number;
                current_number = 0;
                number_exists = 0;
            }
        }
    }

    for(int i = 0; i < result_index; i++)
    {
        printf("%ld ", result[i]);
    }

    return 0;
}

File: Dante1/Module1.05/Zadanie5.11/main.rs
fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).expect("Error");

    for i in input.chars() {
        if !i.is_digit(10) && i != '|' && i != '-' {
            panic!("Incorrect input");
        }
    }

    let mut result: [i64; 100] = [0; 100];
    let mut result_index: usize = 0;
    let mut sum: i64 = 0;

    let mut inside_segment: i32 = 0;
    let mut current_number: i64 = 0;
    let mut number_exists: i32 = 0;

    for i in 0..input.trim().len()
    {
        let mut current = input[i];

        if current == '|' || current == '\0'
        {
            if inside_segment == 1
            {
                sum += current_number;
                result[result_index] = sum;
                result_index += 1;
                sum = 0;
                current_number = 0;
            }
            else
            {
                result[result_index] = 0;
                result_index += 1;
            }
            inside_segment = 1;
            number_exists = 0;
        }
        else if current.isdigit(10)
        {
            current_number = current_number * 10 + (current - '0');
            number_exists = 1;
        }
        else if current == '-'
        {
            if number_exists == 1
            {
                sum = sum + current_number;
                current_number = 0;
                number_exists = 0;
            }
        }
    }


    for i in 0..result_index {
        print!("{} ", result[i]);
    }
}
